drivers/can/Kconfig.stm32

```diff
@@ -9,7 +9,6 @@
 config CAN_STM32
 	bool "STM32 CAN Driver"
 	depends on SOC_FAMILY_STM32
-	select USE_STM32_HAL_CAN
 	help
 	  Enable STM32 CAN Driver (tested on stm32F0 series)
```

Comment:
Is that still right ?

Reply:
Yes. All dependencies from HAL are removed.

Reply:
I meant 'tested on stm32F0 series'

Reply:
This comment is from a very early stage of the driver where we only had support for this CPU.
I think i can safely remove it now.